### ISSUE TITLE
Digits 7 segments. issue

### ISSUE THREAD

**Issue by 7HUND3R36** [SHWotever/SimHub#581]
Why is it displayed in 88888 and nothing else comes out? am I doing something wrong? This is configured on the simhub screen as the 20x4 LCD, right?

https://i.gyazo.com/317a523bfadde1cb921231676e7c07e6.jpg

**Comment by SHWotever** [comment]
Hello ! This is not a 20x4 LCD, this is a max7219/21
![image](https://user-images.githubusercontent.com/2207331/82486917-db8b2280-9add-11ea-917c-42a475686206.png)

**Comment by 7HUND3R36** [comment]
Yes, when I configure the Skecht configured in simhub I select it as max7219 and when loading the library in the arduino nano it shows me 88888888 and it does nothing more than show me that. Can 20x4 LCD and max7219 be connected on simhub at the same time? Could it be due to conflict?

**Comment by SHWotever** [comment]
Yes it can. I would double check wiring if it stays to 8888888 it means that the arduino can't initialize it, probably due to wrong pins order.

**Comment by 7HUND3R36** [comment]
Do I have to follow this mandatory scheme? is that I only have 1 7-digit module. I have it connected as follows directly to the arduino swim

5v - GND
GND - GND
DIN - D3
CS - D4
CLK - D5


54/5000
when I manipulate the 7-digit module it blinks.

**Comment by SHWotever** [comment]
Don't search further ;) if it blinks when manipulating it means there are bad contacts ;)

**Comment by 7HUND3R36** [comment]
Already solved. Don't ask me how to fix it because I don't do anything. The screen just turns on when I am in the single arduino tag on simhub but when i go into multiple arduino it doesn't turn on. Now it shows me the time. I leave you a video, the strange thing is that it no longer interferes when I manipulate the display and the arduino nano does not emit any light?

https://www.youtube.com/watch?v=plsl5aVSE-8

**Comment by SHWotever** [comment]
Hello !

Just let me any other displays you need to set some content. Also make sure multiple Arduino settings are clean and that you only kept currently active devices.https://github.com/SHWotever/SimHub/wiki/Arduino-Multiple-devices-support

**Comment by 7HUND3R36** [comment]
In the tab of multiple arduinos when I execute delete, then after a while LCD appears again. Whenever I add a new component I add it from multiple arduinos.

https://i.gyazo.com/c505a1a829159e195138e8254d98b882.png

**Comment by SHWotever** [comment]
Indeed you simply need to delete the old one  and reorder them. Each time you upload the sketch it will be considered as a new unique device. I had to force this behaviour because lot of people weren't reading the documentation and were giving the same name to multiple device leading to conflicts.

**Comment by 7HUND3R36** [comment]
I already got the LCD and the digits to work at the same time but now I have another problem. In the 2 blocks of 7 digits it shows me the same but in simhub it shows me different each digit block.

https://i.gyazo.com/d32139a5fcf0d6f70ba7519f2e649cd3.png

I have followed this connection scheme. The 2 blocks of digits show the same why are the 2 bridged and why are they cloned? If I want each block of digits to show different parameters, does 1 block of digits have different arduinos?

**Comment by SHWotever** [comment]
Do you have a screenshot of the multiple Arduino settings and if the wiring ?

**Comment by SHWotever** [comment]
The max modules are properly cascaded ? https://github.com/SHWotever/SimHub/wiki/Arduino--MAX7219-MAX7221-7-segment-display#max7219max7221-7-segment-wiring

**Comment by 7HUND3R36** [comment]
YES, the wiring he followed this scheme to the melimeter

https://github.com/SHWotever/SimHub/wiki/Arduino--MAX7219-MAX7221-7-segment-display#max7219max7221-7-segment-wiring

And on simhub I have it on

https://i.gyazo.com/a681fa82c82d2da0f31568a62098786c.png
https://i.gyazo.com/86b968c21a8cbb28f0449254a1da1bb7.png

I only have the 20X4 LCD connected and the 2 7-digit blocks

https://i.gyazo.com/2cf383a30258aa942d9df0c93e98cd6c.jpg

**Comment by SHWotever** [comment]
Are you sure you have set the correct number of modules at upload ? 
I've just checked the max7219 will replicate the content if you set it to 1 module instead of 2.

![image](https://user-images.githubusercontent.com/2207331/82602628-afd96c80-9bb1-11ea-9f38-20900df39d78.png)

**Comment by 7HUND3R36** [comment]
Ah my head !!! therein lies the problem. position 1 nothing else. I am going to try!

**Comment by 7HUND3R36** [comment]
Well there was the problem. I have set it to 2 and now mark the current lap time at the top and the best lap time below.
Thank you very much for your patience and help.

https://www.youtube.com/watch?v=-8AgWTuu4Iw

**Comment by SHWotever** [comment]
They cant be cascaded indeed but they can run on the same arduino as long as you have enough pins, I let you count ;),
You may need an external power : https://github.com/SHWotever/SimHub/wiki/Arduino--Setup#powering-considerations

**Comment by 7HUND3R36** [comment]
The truth is that I want to connect a lot of things but I have few USB ports and 220V plugs. I also don't want to overload the USB. Up to how many 7-digit modules could be cascaded without having to put in an external power supply?
At the moment with the 2 modules I am doing well and they do not do strange things.

**Comment by SHWotever** [comment]
Arduino and standard USB can drive up to 500mA. Hard to tell the consumption of the particular combination you are planning (it varies depending of colour, brightness, etc...), Simply make the sum of each module consumption.

**Comment by 7HUND3R36** [comment]
I had thought to connect this matrix led

https://github.com/SHWotever/SimHub/wiki/Arduino-MAX7219-Matrix-Wiring

To the same arduino where I have the 8-digit modules connected

https://github.com/SHWotever/SimHub/wiki/Arduino--MAX7219-MAX7221-7-segment-display

I bridge the gnd and vcc and then the DIN, CLK, LD I connect them to other pins that I have free of the arduino nano, for example to the d6, d7 and d8 since the d3, d4 and d5 I have them occupied by the 8-digit modules .

Due to configuration issues in simhub you would have no problems right? since I assign pins d6, d7 and d8 to the led matrix and d3, d4 and d5 to the 8-digit module and the gnd and vcc jumper them.

**Comment by penfold42** [comment]
That works - this is what I have running.

Make sure your power and ground wires are good. Those cheap Ebay DuPont wires are TERRIBLE for high current.

If you want to be tricky, you can share some of the IO lines: https://github.com/SHWotever/SimHub/issues/566

**Comment by 7HUND3R36** [comment]
It does work. I have joined the vcc and gnd of the 2 8-digit modules and the led matrix and I have connected them to the arduino nano, then the connectors of the 2 8-digit modules the clk, cs and din I have connected them to the d3, d4 and d5 of the arduino nano and the pins of the led matrix the clk, cs, and din I have connected them to the d6, d7 and d8 of the same arduino nano, then in simhub when configuring the sketch in the same configuration I have chosen the 2 modules of 8 digits with pins d3, d4, d5 that comes by default and I have also added the matrix led but changing the pins that come by default by d6, d7, d8 to connect it to the nano.

Video!

https://youtu.be/nBftzefAqdc
